File: rust_codeforces/cpp/luogu/old/P1059.cpp
/*@author    Allergy
 *@email     Allergy527@gmail.com
 *@workspace luogu\P1059.cpp
 *@data      2024/01/17 19:58:44
 */
#include<bits/stdc++.h>
#define ll long long

using namespace std;

void solve() {
    vector<bool>bucket(1001);
    int n;
    cin >> n;
    int ans = n, i;
    for(int _ = 0;_ < n;++_) {
        cin >> i;
        if(!bucket[i])bucket[i] = true;
        else --ans;
    }
    cout << ans << '\n';
    for(i = 1;i <= 1000;++i)if(bucket[i])cout << i << ' ';
}

int main() {
    ios::sync_with_stdio(false);
    int t = 1;
    //cin >> t;
    while(t--)solve();
    cin >> t;
    return 0;
}

File: rust_luogu/src/bin/Part 2.2/p1059.rs
//@author    Allergy
//@email     Allergy527@gmail.com
//@workspace luogu/p1059.rs
//@data      2024/01/17 23:35:08
fn cin() -> String {
    let mut input = String::new();
    std::io::stdin().read_line(&mut input).unwrap();
    input.trim().to_string()
}
fn main() {
    let t = 1;
    //let mut input = String::new();std::io::stdin().read_line(&mut input).unwrap();let t:i32=input.parse().unwrap();
    (0..t).for_each(|_| solve());
}
fn solve() {
    let mut n: usize = cin().parse().unwrap();
    let mut ans = vec![false; 2799];
    cin().split_whitespace().for_each(|x| {
        let i = x.parse::<usize>().unwrap();
        if ans[i] {
            n -= 1;
        } else {
            ans[i] = true
        };
    });
    println!("{}", n);
    ans.iter().enumerate().for_each(|(i, &x)|if x {print!("{} ", i)});
}
